mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java

```diff
@@ -443,16 +445,19 @@ public void onAvailableConnectionChanged() {
                     mController.indicateMissingConfiguration(false);
                 }
             } else if (failureReason != null) {
-                final String message;
                 if (failureReason instanceof NetworkNotSupportedException) {
                     NetworkInfo info =
                             ((NetworkNotSupportedException) failureReason).getNetworkInfo();
-                    message = getString(R.string.error_network_type_unsupported,
-                            info.getTypeName());
+                    mController.indicateNoNetwork(
+                            getString(R.string.error_network_type_unsupported, info.getTypeName()));
+                } else if (failureReason instanceof NetworkNotAvailableException
+                        && !((WifiManager) getSystemService(Context.WIFI_SERVICE))
```

Comment:
I'd prefer to fetch the WifiManager instance out of the if-else cascade (e.g. in line 448), in order to make the code more clear/readable.